### DISCUSSION TITLE
Errors I found

### DISCUSSION THREAD

**Discussion by xxxhhhyxy** [ViveSoftware/ViveInputUtility-Unity#231] [Q&A]
in the Dragable.cs, 
1. how do you define the 3 public "UnityEventDraggable": afterGrabbed, beforeRelease and onDrop? Because I notice no difference between the latter two. I used to think that beforeRelease is for the update of the event.  
and now could you please add an UnityEventDraggable for updating?
2. for the input parameter of the 3 events above, the Pos offset is always zero, I don't know why.

**Comment by lawwong** [comment]
1.1 beforeRelease and onDrop behave differently when there are multiple grabbers. Consider following sequence:
      * **Grabber1** grabs
      * afterGrabberGrabbed invoked (this.currentGrabber returns **Grabber1** instance)
      * **Grabber2** grabs
      * afterGrabberGrabbed invoked (this.currentGrabber returns **Grabber2** instance)
      * **Grabber2** release
      * beforeGrabberReleased invoked just before **Grabber2** acutally release (hence this.currentGrabber returns **Grabber2** instance)
      * **Grabber1** release
      * beforeGrabberReleased invoked just before **Grabber1** acutally release (hence this.currentGrabber returns **Grabber1** instance)
      * onGrabberDrop invoked since no grabber is holding (this.currentGrabber returns null)

1.2 About "add UnityEventDraggable event for updating", will add to todo list

2. The posOffset field is inherit from the base class BasePoseTracker. It means tracking offset to the final grabbed position. Use currentGrabber.grabberOrigin & currentGrabber.grabOffset to get grabber's information.
